web/src/message_edit_history.ts

```diff
@@ -50,81 +84,86 @@ export function fetch_and_render_message_history(message) {
-                    item.edited_by_notice = $t(
-                        {defaultMessage: "Moved by {full_name}"},
-                        {full_name},
-                    );
-                    item.topic_edited = true;
-                    item.prev_topic = msg.prev_topic;
-                    item.new_topic = msg.topic;
+                    edited_by_notice = $t({defaultMessage: "Moved by {full_name}"}, {full_name});
+                    topic_edited = true;
+                    prev_topic = msg.prev_topic;
+                    new_topic = msg.topic;
                 } else if (msg.prev_stream) {
                     const sub = sub_store.get(msg.prev_stream);
-                    item.edited_by_notice = $t(
-                        {defaultMessage: "Moved by {full_name}"},
-                        {full_name},
-                    );
-                    item.stream_changed = true;
+                    edited_by_notice = $t({defaultMessage: "Moved by {full_name}"}, {full_name});
+                    stream_changed = true;
                     if (!sub) {
-                        item.prev_stream = $t({defaultMessage: "Unknown stream"});
+                        prev_stream = $t({defaultMessage: "Unknown stream"});
                     } else {
-                        item.prev_stream = sub_store.maybe_get_stream_name(msg.prev_stream);
+                        prev_stream = sub_store.maybe_get_stream_name(msg.prev_stream);
                     }
                     if (prev_stream_item !== null) {
                         prev_stream_item.new_stream = sub_store.maybe_get_stream_name(
                             msg.prev_stream,
                         );
                     }
-                    prev_stream_item = item;
                 } else {
                     // just a content edit
-                    item.edited_by_notice = $t(
-                        {defaultMessage: "Edited by {full_name}"},
-                        {full_name},
-                    );
-                    item.body_to_render = msg.content_html_diff;
+                    edited_by_notice = $t({defaultMessage: "Edited by {full_name}"}, {full_name});
+                    body_to_render = msg.content_html_diff;
+                }
+
+                const item = {
+                    timestamp,
+                    display_date,
+                    show_date_row,
+                    edited_by_notice,
+                    body_to_render,
+                    topic_edited,
+                    prev_topic,
+                    new_topic,
+                    stream_changed,
+                    prev_stream,
+                };
+
+                if (msg.prev_stream) {
+                    prev_stream_item = item;
                 }
 
                 content_edit_history.push(item);
                 prev_time = time;
             }
-            if (prev_stream_item !== null) {
+            if (prev_stream_item !== null && message.type === "stream") {
```

Comment:
Why this change?

Reply:
`message` only has a `stream_id` if it's type stream, so it's needed for typescript. Previously, if a private message got through here then `maybe_get_stream_name` would return `undefined`, so `new_stream` would be set to `undefined`, which is the same effect as just not entering the `if` statement at all.

Reply:
Probably a better case for an assert.